hibernate-core/src/main/java/org/hibernate/id/uuid/UuidV7ValueGenerator.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.id.uuid;
+
+import java.security.SecureRandom;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.id.UUIDGenerationStrategy;
+
+import static java.time.Instant.EPOCH;
+import static java.time.temporal.ChronoUnit.MILLIS;
+
+/**
+ * Implements a <a href="https://datatracker.ietf.org/doc/html/rfc9562#name-uuid-version-7">UUID Version 7</a> generation strategy as defined by the {@link UUID#randomUUID()} method.
+ *
+ * @author Cedomir Igaly
+ */
+public class UuidV7ValueGenerator implements UUIDGenerationStrategy, UuidValueGenerator {
+
+	public static final UuidV7ValueGenerator INSTANCE = new UuidV7ValueGenerator();
+
+	private static class Holder {
+
+		static final SecureRandom numberGenerator = new SecureRandom();
+	}
+
+	private final Lock lock = new ReentrantLock( true );
+
+	private Duration currentTimestamp;
+
+	private final AtomicLong clockSequence;
+
+	public UuidV7ValueGenerator() {
+		this( getCurrentTimestamp(), 0 );
+	}
+
+	public UuidV7ValueGenerator(final Duration currentTimestamp, final long clockSequence) {
+		this.currentTimestamp = currentTimestamp;
+		this.clockSequence = new AtomicLong( clockSequence );
+	}
+
+	/**
+	 * A variant 7
+	 */
+	@Override
+	public int getGeneratedVersion() {
+		/*
+		 *	UUIDv7 features a time-ordered value field derived from the widely implemented and well-
+		 * known Unix Epoch timestamp source, the number of milliseconds since midnight 1 Jan 1970 UTC,
+		 * leap seconds excluded.
+		 */
+		return 7;
+	}
+
+	/**
+	 * Delegates to {@link UUID#randomUUID()}
```

Comment:
Not sure why you are referencing `UUID#randomUUID` here?

Reply:
Same mistake as on other three places